core/src/Cuda/Kokkos_Cuda_Half_Conversion.hpp

```diff
@@ -313,7 +317,11 @@ KOKKOS_INLINE_FUNCTION
 #endif
 
 /************************** bhalf conversions *********************************/
-#if CUDA_VERSION >= 11000 && CUDA_VERISON < 11200
+// Go in this branch if CUDA version is >= 11.0.0 and less than 11.1.0 or if the
+// architecture is not Ampere
+#if CUDA_VERSION >= 11000 && \
+    (CUDA_VERSION < 11010 || \
+     (!defined(KOKKOS_ARCH_AMPERE80) && !defined(KOKKOS_ARCH_AMPERE82)))
```

Comment:
`KOKKOS_ARCH_AMPERE82` does not exist but in any case, any reason not to use `!defined(KOKKOS_ARCH_AMPERE80)`?

Reply:
You also need to get in this branch with Ampere with some versions of CUDA. The functions exists when using Ampere but they are device only. In the newer versions of CUDA they are host device.